### PR TITLE
[IMPROVEMENT] Adding Webhook handler for Gogs/Gitea

### PR DESCRIPTION
**Pull request by axeloz**
**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributing guidelines](https://github.com/REBElinBLUE/deployer/.github/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] Do the TravisCI tests pass?
- [x] Does the StyleCI test pass?
---

### Description of change
Adding a Gogs/Gitea handler for Webhook. 

*Please provide a description of the change here.*
Github handler doesn't work for Gogs as the "head_commit" doesn't exist in the Payload.

### CODE REVIEW

**Comment by axeloz** [comment]
Fixes issue #288

**Comment by REBELinBLUE** [comment]
👍 I'll get it merged this evening and tag a new release also. One minor thing, would you be able to update https://github.com/REBELinBLUE/deployer/blob/master/resources/views/projects/dialogs/webhook.blade.php#L46 to list Gogs/Gitea as well

File: app/Http/Webhooks/Gogs.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace REBELinBLUE\Deployer\Http\Webhooks;
+
+/**
+ * Class to handle integration with Github webhooks.
```
Comment (axeloz) [inline app/Http/Webhooks/Gogs.php]: Pushed too fast earlier, let mentions of Github in comments, dirty work, sorry

**Comment by REBELinBLUE** [comment]
Oops, and I didn't notice that either 🙊

Stephen Ball
stephen@rebelinblue.com

**Pull request merged**